src/main/webapp/script.js

```diff
@@ -0,0 +1,382 @@
+
+  if (!SUPPORTED_EXTENSIONS.includes(fileExtension)) {
+    alert('We can support only jpg, jpeg and png images, sorry :(');
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Creates canvas with specified id.
+ * @param {string} id
+ * @return {HTMLCanvasElement} canvas DOM element
+ */
+function createCanvas(id) {
+  const canvas = document.createElement('canvas');
+  canvas.setAttribute('id', id);
+
+  return canvas;
+}
+
+/**
+ * Creates canvas with width and height of an image
+ * @param {Image} image
+ * @return {HTMLCanvasElement} canvas DOM element
+ */
+function createCanvasForImage(image) {
+  const canvas = document.createElement('canvas');
+  canvas.setAttribute('width', image.width);
+  canvas.setAttribute('height', image.height);
+
+  return canvas;
+}
+
+/**
+ * Function to handle all the blurring process
+ * and putting both blurred and original images on page
+ * @param {string} imageUrl
+ */
+async function blur(imageUrl) {
+  // get areas to blur from server
+  const rectsToBlurPromise = getBlurAreas();
+
+  // create Imge object from url to put it on canvases
+  const imageObj = new Image();
+  imageObj.src = imageUrl;
+
+  // need to wait until image loads to put it anywhere
+  imageObj.onload = () => {
+    prepareHtmlElements(imageObj);
+
+    // wait for server to respond and finish blurring
+    rectsToBlurPromise.then((rects) => {
+      const blurredImage = getBlurredImage(rects, imageObj);
+      createDownloadButton(blurredImage);
+      showBlurredImage(blurredImage);
+    });
+  };
+}
+
+/**
+ * Fucntion to update html page for blurring a new image.
+ * It resizes canvases and draws new image on them.
+ * It deletes previous download button if there was.
+ * @param {Image} image
+ */
+function prepareHtmlElements(image) {
+  const prepareCanvas = (canvasId) => {
+    const canvas = document.getElementById(canvasId);
+    const ctx = canvas.getContext('2d');
+
+    // clear canvas
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
+
+    // resize canvas height to fit new image
+    canvas.height = image.height * canvas.width / image.width;
+
+    // draw new image on it, scaling the image to fit in canvas
+    ctx.drawImage(image, 0, 0, canvas.width, canvas.height);
+  };
+  prepareCanvas('input-canvas');
+  prepareCanvas('output-canvas');
+
+  // remove download button
+  const downloadButton = document.getElementById('download-button');
+  if (downloadButton !== null) {
+    downloadButton.remove();
+  }
+}
+
+/**
+ * Class for a rectangle to blur
+ */
+class Rect {
+  /**
+   * @param {function} func Can be min or max function
+   * @param {string} property Property which all the items
+   *                          in arrayObject have
+   * @param {Array} arrayObject array of Objects
+   * @return {Number} min or max (depending on func) value of the
+   *         specified property among arrayObject items.
+   */
+  static get(func, property, arrayObject) {
```

Comment:
It took me a while to understand what this function does. Maybe a better naming and a comment explaining it?